plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftSparkMetastoreUtil.java

```diff
@@ -0,0 +1,166 @@
+        OptionalDouble avgLength = toDouble(columnStatistics.get(field + "avgLen"));
+        OptionalLong nullsCount = toLong(columnStatistics.get(field + "nullCount"));
+        OptionalLong distinctValuesCount = toLong(columnStatistics.get(field + "distinctCount"));
+        if (tableRowCount.isPresent() && partitionRowCount.isPresent() && !tableRowCount.equals(partitionRowCount)) {
+            // Spark does not store statistics for partitions, but only for table/column.
+            nullsCount = OptionalLong.empty();
+            distinctValuesCount = OptionalLong.of(Math.min(distinctValuesCount.getAsLong(), partitionRowCount.getAsLong()));
+        }
+
+        return switch (((PrimitiveTypeInfo) typeInfo).getPrimitiveCategory()) {
+            case BOOLEAN -> createBooleanColumnStatistics(
+                    OptionalLong.empty(),
+                    OptionalLong.empty(),
+                    nullsCount);
+            case BYTE, SHORT, INT, LONG -> createIntegerColumnStatistics(
+                    toLong(columnStatistics.get(field + COLUMN_MIN)),
+                    toLong(columnStatistics.get(field + COLUMN_MAX)),
+                    nullsCount,
+                    fromMetastoreDistinctValuesCount(distinctValuesCount, nullsCount, partitionRowCount));
+            case TIMESTAMP -> createIntegerColumnStatistics(
+                    OptionalLong.empty(),
+                    OptionalLong.empty(),
+                    nullsCount,
+                    fromMetastoreDistinctValuesCount(distinctValuesCount, nullsCount, partitionRowCount));
+            case FLOAT, DOUBLE -> createDoubleColumnStatistics(
+                    toDouble(columnStatistics.get(field + COLUMN_MIN)),
+                    toDouble(columnStatistics.get(field + COLUMN_MAX)),
+                    nullsCount,
+                    fromMetastoreDistinctValuesCount(distinctValuesCount, nullsCount, partitionRowCount));
+            case STRING, VARCHAR, CHAR -> createStringColumnStatistics(
+                    maxLength,
+                    getTotalSizeInBytes(avgLength, partitionRowCount, nullsCount),
+                    nullsCount,
+                    fromMetastoreDistinctValuesCount(distinctValuesCount, nullsCount, partitionRowCount));
+            case DATE -> createDateColumnStatistics(
+                    toDate(columnStatistics.get(field + COLUMN_MIN)),
+                    toDate(columnStatistics.get(field + COLUMN_MAX)),
+                    nullsCount,
+                    fromMetastoreDistinctValuesCount(distinctValuesCount, nullsCount, partitionRowCount));
+            case BINARY -> createBinaryColumnStatistics(
+                    maxLength,
+                    getTotalSizeInBytes(avgLength, partitionRowCount, nullsCount),
+                    nullsCount);
+            case DECIMAL -> createDecimalColumnStatistics(
+                    toDecimal(columnStatistics.get(field + COLUMN_MIN)),
```

Comment:
optional: Would it make sense to have a test which makes sure that illegal values in the stats do not cause the `SHOW STATS` statement to fail?

Reply:
We could have such tests, but from my point of view it would be artificial.